### DISCUSSION TITLE
Problem with hostid

### DISCUSSION THREAD

**Discussion by zarMarco** [zbm-dev/zfsbootmenu#371] [General]
Hi all, I've a question/problem. I have one zfs partition, one zpool, that contain dataset for arch/void/opensuse and freebsd. For booting linux distros I use zfsbootmenu ad this is fantastic. But when I booted freebsd at reboot I found hostid 000000 on linux distros and void and opensuse cannot boot. I've partialy solved with skip on recovery shell of zfsbootmenu and export pool, write hostid and import pool. On freebsd I've setted same hostid of linux. How can I solve this?

**Comment by ahesford** [comment]
What kernel command line shows up at the top of your ZFSBootMenu screen before you try to boot but fail?

**Comment by zarMarco** [comment]
On arch line I've "rpool/arch-gnome/root (r/o) - vmlinuz-linux rw spl_hostid=00000000"

Can I add on /etc/zfsbootmenu/config.yaml on kernel line the correct hostid?

**Comment by ahesford** [comment]
Setting `spl_hostid=00000000` on the command-line is a last-ditch effort by ZFSBootMenu, but this is known to work only with dracut. Arch's `mkinitcpio` will not respect `spl_hostid` values.

The reason ZBM is trying to set `spl_hostid` instead of the module-level `spl.spl_hostid` is that the module interprets a zero value as "get the hostid from `/etc/hostid`".

Generally, `hostid` is a concept that is more trouble than it's worth. We've spent a lot of effort trying to work around some of the warts, but some are unavoidable. Your only shot here is to make sure all of your systems share a hostid. You can also try setting `zbm.import_policy=force` and make sure that the ZFSBootMenu image has the expected value in `/etc/hostid`.

I'm kicking this to discussions, as it isn't really a ZFSBootMenu issue.

**Comment by zarMarco** [comment]
But I've same problem with void and zfsbootmenu is installed on void. in fact arch is only distro that boot correctly although spl_hostid is 0000000

**Comment by zdykstra** [comment]
Double check the value of hostid across every OS that you're booting. One of them seems to be importing the pool with a hostid of '0'. Until they're all using the exact same hostid, you're going to run into issues; especially if one has a hostid of '0'. That's a horrible edge case for ZFS on Linux.

**Comment by zarMarco** [comment]
ok, when distros boot hostid is correct. This is ok eh eh
